src/graph/search.cc

```diff
@@ -696,7 +696,7 @@ ncclResult_t ncclTopoSearchRecNet(struct ncclTopoSystem* system, struct ncclTopo
             if (paths[i].count <= paths[f].count) {
               // prefer GPU direct RDMA
               enum ncclTopoGdrMode useGdr;
-              NCCLCHECK(ncclTopoCheckGdr(system, system->nodes[GPU].nodes[i].id, net->id, 0, &useGdr));
+              NCCLCHECK(ncclTopoCheckGdr(system, system->nodes[GPU].nodes[i].gpu.rank, net->id, 0, &useGdr));
```

Comment:
It doesn't make sense to me to compare the rank id to NIC bus ID. This will break in many cases. Need to check why this failed for the particular pattern rather than changing the comparison terms

Reply:
Looks like the implementation of the underlying function has changed to map rank to index using `ncclTopoRankToIndex`. With this modification, passing rankId is no longer a choice. Please update the PR title to something like: "[GRAPH] Pass rank instead of busID due to a change in an internal function signature"